roxie/ccd/ccdlistener.cpp

```diff
@@ -945,6 +945,61 @@ extern void updateAffinity(unsigned __int64 affinity)
 
 //--------------------------------------------------------------------------------------------------------------------
 
+StringBuffer & ContextLogger::getStats(StringBuffer &s) const
+{
+    CriticalBlock block(statsCrit);
+    stats.toStr(s);
+
+    if (slowActivityIds[0])
+    {
+        StringBuffer ids;
+        StringBuffer times;
+        for (unsigned i=0; i < MaxSlowActivities; i++)
+        {
+            if (!slowActivityIds[i])
+                break;
+
+            if (i)
+            {
+                ids.append(",");
+                times.append(",");
+            }
+            ids.append(slowActivityIds[i]);
+            formatStatistic(times, cycle_to_nanosec(slowActivityTimes[i]), SMeasureTimeNs);
+        }
+        s.appendf(", slowActivities={ ids=[%s] times=[%s] }", ids.str(), times.str());
```

Comment:
Not "slowActivities" but perhaps "slowestActivities" ?
What if they are all about the same ?  Is there more we can provide ?  As in:
complete in 11861 msecs ...
slowActivities={ ids=[45,195,75,63,15] times=[10.196s,10.059s,10.035s,10.028s,10.021s] }